### PR TITLE
[Contributing] Fix Coding Standards example (esp. null return vs. void)

### PR DESCRIPTION
**Pull request by guilliamxavier**
Fix inconsistent return points and make the logic a bit more realistic (even if still dummy code)

### CODE REVIEW

File: contributing/code/standards.rst [hunk 1]
```diff
@@ -101,33 +109,34 @@ short example containing most features described below::
             );
 
             if (true === $dummy) {
-                return null;
+                return 'something';
```
Comment (guilliamxavier) [inline contributing/code/standards.rst]: Note: alternatively you could keep `return null;` here and do `return 'something';` at the end, but feels "weird" to me

**Comment by guilliamxavier** [comment]
What's the matter with the "DOCtor-RST" lint failing?

**Comment by javiereguiluz** [comment]
@guilliamxavier you can ignore the DOCtor-RST linter errors for this pull request.

About the proposed changes, let's ask @nicolas-grekas and @stof for a review. Thanks!

File: contributing/code/standards.rst [hunk 2]
```diff
@@ -89,8 +97,8 @@ short example containing most features described below::
                 'another_default' => 'more values',
             ];
 
-            foreach ($options as $option) {
-                if (!in_array($option, $defaultOptions)) {
+            foreach (array_keys($options) as $option) {
```
Comment (guilliamxavier) [inline contributing/code/standards.rst]: not sure if you prefer `foreach ($options as $option => $unused) {`

File: contributing/code/standards.rst [hunk 3]
```diff
@@ -101,33 +109,34 @@ short example containing most features described below::
             );
 
             if (true === $dummy) {
-                return null;
+                return 'something';
             }
 
-            if ('string' === $dummy) {
+            if (is_string($dummy)) {
```
Comment (guilliamxavier) [inline contributing/code/standards.rst]: Note: I have seen `\is_string(` (backslash-prepended) in the code but not so in the docs

File: contributing/code/standards.rst [hunk 4]
```diff
@@ -101,33 +109,34 @@ short example containing most features described below::
...
 
         /**
-         * Performs some basic check for a given value.
+         * Performs some basic operations for a given value.
          *
-         * @param mixed $value     Some value to check against
+         * @param mixed $value     Some value to operate against
          * @param bool  $theSwitch Some switch to control the method's flow
-         *
-         * @return bool|void The resultant check if $theSwitch isn't false, void otherwise
```
Comment (guilliamxavier) [inline contributing/code/standards.rst]: Note: I have seen *some* `@return void` in the code, but below is explicitly stated
> Omit the ``@return`` tag if the method does not return anything

File: contributing/code/standards.rst [hunk 5]
```diff
@@ -52,12 +54,18 @@ short example containing most features described below::
          */
         private $fooBar;
 
+        /**
+         * @var Qux
```
Comment (nicolas-grekas) [inline contributing/code/standards.rst]: actually, when the constructor already hints about the type, we recommend not duplicating in a code block.

File: contributing/code/standards.rst (hunk 2 above)
Comment (nicolas-grekas) [reply contributing/code/standards.rst]: `foreach ($options as $name => $values)` would prevent an array copy

**Review by nicolas-grekas — COMMENTED** [review]
Just one minor note and GTM

**Comment by guilliamxavier** [comment]
@nicolas-grekas: Thank you for reviewing, I have applied your suggested changes

File: contributing/code/standards.rst (hunk 3 above)
Comment (OskarStark) [reply contributing/code/standards.rst]: This not the same anymore before it did not check if it is a string, but if the string is the same string as `string`....

Comment (guilliamxavier) [reply contributing/code/standards.rst]: That's intended, as I said:

> [...] and make the logic a bit more realistic (even if still dummy code)

Look at the code inside that `if`, before it was basically either `return substr('string', 0, 5);` or `return ucwords('string');`...

**Review by OskarStark — APPROVED** [review]

**Comment by javiereguiluz** [comment]
Thank you @guilliamxavier.

**Pull request merged**